objects/Editors/contained/3d8fa9/contained/0e35be/script.lua

```diff
@@ -0,0 +1,143 @@
+        local paddingX = 1.1
+        for i = 1,8 do
+            self.createButton({
+                click_function = "button"..i,
+                function_owner = self,
+                label = ">",
+                rotation = {0,90,0},
+                font_size = 500,
+                color = hexToDec(elementColors[i]),
+                font_color = {0,0,0},
+                position = {0.5+i*paddingX,0.1,1},
+                width = 400,
+                height = 500,
+            })
+            local func = function() editThreshold(i,-1) end
+            self.setVar("button"..i,func)
+
+            self.createButton({
+                click_function = "nullFunc",
+                function_owner = self,
+                label = threshold[i],
+
+                font_size = 500,
+                color = hexToDec(elementColors[i]),
+                font_color = {0,0,0},
+                position = {0.5+i*paddingX,0.1,0},
+                width = 500,
+                height = 500,
+            })
+
+            self.createButton({
+                click_function = "button"..i+8,
+                function_owner = self,
+                label = "<",
+                rotation = {0,90,0},
+                font_size = 500,
+                color = hexToDec(elementColors[i]),
+                font_color = {0,0,0},
+                position = {0.5+i*paddingX,0.1,-1},
+                width = 400,
+                height = 500,
+            })
+            func = function() editThreshold(i,1) end
+            self.setVar("button"..i+8,func)
+        end
+        self.createButton({
+            click_function = "changeButtons",
+            function_owner = self,
+            label = "Hide",
+	    rotation = {0,270,0},
+            font_size = 400,
+            color = {1,1,1},
+            font_color = {0,0,0},
+            position = {-1.6,0.1,0},
+            width = 1000,
+            height = 500,
+        })
+    else
+        self.clearButtons()
+        self.createButton({
+            click_function = "changeButtons",
+            function_owner = self,
+            label = "Show",
+	    rotation = {0,270,0},
+            font_size = 400,
+            color = {1,1,1},
+            font_color = {0,0,0},
+            position = {-1.6,0.1,0},
+            width = 1000,
+            height = 500,
+        })
+    end
+end
+
+function editThreshold(eleIndex, num)
+    threshold[eleIndex] = math.max(threshold[eleIndex]+num,0)
+    self.script_state = JSON.encode({
+        threshold = threshold,
+        editMode = editMode
+    })
+    updateButtons()
+end
+
+function changeButtons()
+    editMode = not editMode
+    self.script_state = JSON.encode({
+        threshold = threshold,
+        editMode = editMode
+    })
+    updateButtons()
+end
+
+function nullFunc()
+end
+
+function hexToDec(inp)
+    inp = string.lower(inp)
+    local colors = {}
+    colors["R"] = string.sub(inp,1,1)
+    colors["r"] = string.sub(inp,2,2)
```

Comment:
why do you have both upper and lower case values?

Reply:
This was just a function used elsewhere, so I also used it here